### ISSUE TITLE
Snippets for custom css-counters

### ISSUE THREAD

**Issue by ericscheid** [naturalcrit/homebrewery#2829]
### Your idea:

From DOMT
> Also, in a similar vein, references. In overleaf there are a set of commands to reference an item in the document (like a paragraph, equation, picture, or even a section). So it would be incredibly useful to have that as well. Making it easier to navigate through the contents.
>
>To avoid confusion, what I was referring to [here] was the `\label{}`, `\cite{}` and `\ref{}` functions overleaf has. HB has `[{{_Text}}](#p)`, which returns an hyperlink Text that doesn't add the number of reference (like "fig. 10" which is the overleaf format).

It is possible to use custom css to insert an incrementing reference number (e.g. `##### Table {tableCounter }: Misc. Stuff`). 

It would be helpful to have some snippets that show what the format should be. Snippets for "Incrementing Table number", or "Incrementing Figure reference" .. with the necessary css already defined in theme-style.css. (Some theme sources even have style conventions of prefixing all tables with "Table NN: ").

If we define this custom css in THB, we can also detect it when rendering and insert `window.getComputedStyle(elem).counter(counter-name)` for a limited but known set of counter-names (e.g. `tableCounter`, `figureCounter`). Those IDs could then be used to generate a TOC-like list of tables or figures.

Caveat: the links of such a TOC-like list of tables or figures would fail if partial-rendering is active for the document (only the page stubs are rendered, not the page contents, for pages which are not the currently viewed or adjacent thereof). Would still be helpful for `/print` resources. (partially rendering fragment targets could be a separate issue).

**Comment by dbolack-ab** [comment]
At this stage, would this not be better handled by the impending variables system?

**Comment by ericscheid** [comment]
> At this stage, would this not be better handled by the impending variables system?

Definitely, especially since the values of `css-counter` can not be accessed by javascript. They can only be displayed according to css (and reset and incremented). This would make using `css-counter` impossible for the use case of "See figure [%css-counter-foo%]".

Using variables we would instead increment the variable and then assign the value of that counting variable to a nominative variable (e.g. increment $[figure_counter], assign $[figure_counter] to $[figure_dog], and use $[figue_dog] eveytime we want to display the value figure_counter had _at the time_ we incremented the counter for the dog figure).

**Comment by 5e-Cleric** [comment]
removing phb style label, since this has a use-case for every theme

**Comment by dbolack-ab** [comment]
Can we reword this problem assuming a brew variables based solution?
